L1Trigger/L1THGCal/src/backend/HGCalMulticlusteringHistoImpl.cc

```diff
@@ -326,40 +458,69 @@ std::vector<GlobalPoint> HGCalMulticlusteringHistoImpl::computeThresholdSeeds( c
 
 
 std::vector<l1t::HGCalMulticluster> HGCalMulticlusteringHistoImpl::clusterSeedMulticluster(const std::vector<edm::Ptr<l1t::HGCalCluster>> & clustersPtrs,
-											   const std::vector<GlobalPoint> & seeds){
-
+                                                                                           const std::vector<std::pair<GlobalPoint, double> > & seeds){
 
     std::map<int,l1t::HGCalMulticluster> mapSeedMulticluster;
     std::vector<l1t::HGCalMulticluster> multiclustersTmp;
 
     for(auto & clu : clustersPtrs){
 
-
         int z_side = triggerTools_.zside(clu->detId());
 
-        double minDist = dr_;
-        int targetSeed = -1;
+        double minDist = drA_ + drB_*(2.3 - std::abs(clu->eta()) ) ;
 
-        for( unsigned int iseed=0; iseed<seeds.size(); iseed++ ){
+        std::vector<int> targetSeeds;
+        std::vector<double> targetSeedsEnergy;
 
-            if( z_side*seeds[iseed].z()<0) continue;
+        for( unsigned int iseed=0; iseed<seeds.size(); iseed++ ){
 
-            double d = this->dR(*clu, seeds[iseed]);
+            if( z_side*seeds[iseed].first.z()<0) continue;
+            double d = this->dR(*clu, seeds[iseed].first);
+
+            if ( d < minDist ){
+              if ( cluster_association_ == "energy" ){
```

Comment:
Should use enum instead of string comparisons.
Why is it called "energy"? There seem to be no energy involved.

Reply:
This is a clustering strategy that allows a hit to be assigned to multiple seeds, based on the fraction of total energy (within a certain radius) each seed has. In the first step a list of seeds within the radius is compiled and then in the second step the splitting is performed.